be/src/vec/exec/volap_scan_node.h

```diff
@@ -28,24 +28,26 @@ namespace vectorized {
 
 class VOlapScanner;
 
-class VOlapScanNode : public OlapScanNode {
+class VOlapScanNode final : public OlapScanNode {
 public:
     VOlapScanNode(ObjectPool* pool, const TPlanNode& tnode, const DescriptorTbl& descs);
-    ~VOlapScanNode();
-    virtual void transfer_thread(RuntimeState* state);
-    virtual void scanner_thread(VOlapScanner* scanner);
-    virtual Status get_next(RuntimeState* state, RowBatch* row_batch, bool* eos) {
+    friend class VOlapScanner;
+
+    void transfer_thread(RuntimeState* state);
+    void scanner_thread(VOlapScanner* scanner);
+    Status get_next(RuntimeState* state, RowBatch* row_batch, bool* eos) override {
         return Status::NotSupported("Not Implemented VOlapScanNode Node::get_next scalar");
     }
-    virtual Status get_next(RuntimeState* state, Block* block, bool* eos);
-    virtual Status add_one_block(Block* block);
-    virtual Status start_scan_thread(RuntimeState* state);
-    virtual Status close(RuntimeState* state);
-
-    friend class VOlapScanner;
+    Status get_next(RuntimeState* state, Block* block, bool* eos) override;
+    Status add_blocks(std::vector<Block*>& block);
```

Comment:
i suggests you distinguish public interface from private implements, make the number of interface min

Reply:
ok